### DISCUSSION TITLE
Build issues: Gradle versions, paths, and SearchUICoreGenericTest.java:23?

### DISCUSSION THREAD

**Discussion by Fmstrat** [osmandapp/OsmAnd#12681] [Q&A]
Hi,

I'm having a number of build issues using Android Studio. I clone the repo into `OsmAnd`, and I run into:
```
Your project contains 2 or more modules with the same identification net.osmand:OsmAnd
     at ":" and ":OsmAnd"
```
This appears to be related to Gradle, where you can do this in older version (1.2.3) but not newer versions (1.3.1) (https://stackoverflow.com/questions/33530050/upgrade-issue-in-gradle-build-tools-version-from-1-2-3-to-1-3-1). Is an older version of Gradle being used?

I then try to rename the root folder from `OsmAnd` to `OsmAndTest` to get around this limitation, but I then get:
```
> Task :OsmAnd-java:test FAILED

net.osmand.search.SearchUICoreGenericTest > classMethod FAILED
    java.lang.RuntimeException at SearchUICoreGenericTest.java:23
        Caused by: java.io.FileNotFoundException at SearchUICoreGenericTest.java:23

net.osmand.router.RouteTestingTest > initializationError FAILED
    java.lang.NullPointerException at RouteTestingTest.java:42

net.osmand.router.RouteResultPreparationTest > initializationError FAILED
    java.lang.NullPointerException at RouteResultPreparationTest.java:76

22 tests completed, 3 failed

        Caused by: java.io.FileNotFoundException at SearchUICoreGenericTest.java:23
```

So it looks like the tests are failing because it can't find the files? I go into the `Settings->Build-Compiler` options in Android Studio and add a Gradle flag of `-x test` to skip them, and then I also add `org.gradle.jvmargs=-Xmx4098m` to `gradle.properties` because it keeps giving heap space errors. 

I get closer, but the paths still seem off, with this coming next:
```
ERROR:/home/fmstrat/Documents/Repos/OsmAndDev/OsmAnd/res/layout/data_storage_list_item.xml:32: AAPT: error: resource drawable/mm_storage_tank (aka net.osmand:drawable/mm_storage_tank) not found.
```
Is there something I need to be doing to get paths to work for the tests and drawable objects?

Thanks!

**Comment by vshcherb** [comment]
Probably you didn't checkout repo correctly - https://docs.osmand.net/en/main@latest/development/build-osmand/setup-the-dev-environment
